vizro-core/tests/unit/vizro/models/_components/form/test_date_picker.py

```diff
@@ -0,0 +1,175 @@
+except ImportError:  # pragma: no cov
+    from pydantic import ValidationError
+
+import vizro.models as vm
+
+
+class TestDatePickerInstantiation:
+    """Tests model instantiation."""
+
+    def test_create_datepicker_mandatory(self):
+        date_picker = vm.DatePicker()
+
+        assert hasattr(date_picker, "id")
+        assert date_picker.type == "date_picker"
+        assert date_picker.min is None
+        assert date_picker.max is None
+        assert date_picker.value is None
+        assert date_picker.title == ""
+        assert date_picker.actions == []
+        assert date_picker.range is True
+
+    def test_create_datepicker_mandatory_and_optional(self):
+        date_picker = vm.DatePicker(
+            id="date-picker-id", min="2024-01-01", max="2024-12-31", value=["2024-03-01", "2024-04-01"], title="Title"
+        )
+
+        assert date_picker.id == "date-picker-id"
+        assert date_picker.type == "date_picker"
+        assert date_picker.min == date(2024, 1, 1)
+        assert date_picker.max == date(2024, 12, 31)
+        assert date_picker.value == [date(2024, 3, 1), date(2024, 4, 1)]
+        assert date_picker.title == "Title"
+        assert date_picker.actions == []
+        assert date_picker.range is True
+
+    @pytest.mark.parametrize("title", ["test", 1, 1.0, """## Test header""", ""])
+    def test_valid_title(self, title):
+        date_picker = vm.DatePicker(title=title)
+
+        assert date_picker.title == str(title)
+
+    def test_set_action_via_validator(self, identity_action_function):
+        date_picker = vm.DatePicker(actions=[vm.Action(function=identity_action_function())])
+        actions_chain = date_picker.actions[0]
+
+        assert actions_chain.trigger.component_property == "value"
+
+    @pytest.mark.parametrize("min, max", [("2024-01-01", None), (None, "2024-01-01"), ("2024-01-01", "2024-02-01")])
+    def test_valid_min_max(self, min, max):
+        date_picker = vm.DatePicker(min=min, max=max)
+
+        assert date_picker.min == (datetime.strptime(min, "%Y-%m-%d").date() if min else None)
+        assert date_picker.max == (datetime.strptime(max, "%Y-%m-%d").date() if max else None)
+
+    def test_validate_max_invalid_min_greater_than_max(self):
+        with pytest.raises(
+            ValidationError, match="Maximum value of component is required to be larger than minimum value."
+        ):
+            vm.DatePicker(min="2024-02-01", max="2024-01-01")
+
+    def test_validate_max_invalid_date_format(self):
+        with pytest.raises(ValidationError, match="invalid date format"):
+            vm.DatePicker(min="50-50-50", max="50-50-50")
+
+    @pytest.mark.parametrize(
+        "range, value",
+        [
+            (False, "2024-01-01"),
+            (False, date(2024, 1, 1)),
+            (True, ["2024-01-01", "2024-02-01"]),
+            (True, [date(2024, 1, 1), date(2024, 2, 1)]),
+        ],
+    )
+    def test_validate_datepicker_value_valid(self, range, value):
```

Comment:
This test is pretty confusing to read because it's handling both the `range=True` case and the `range=False` case in one go and these things can return a list or a single value depending on the parameter.

Also I don't know what the point of testing both `"2024-01-01"` and `date(2024, 1, 1)` is? Isn't that just testing that pydantic is coercing `value` to `date`?

Personally I would simplify this into two tests:
```
def test_validate_datepicker_value_single_valid():
    date_picker = vm.DatePicker(min="2024-01-01", max="2024-02-01", range=False, value=...)
    assert date_picker.value == ...

def test_validate_datepicker_value_range_valid():
    date_picker = vm.DatePicker(min="2024-01-01", max="2024-02-01", range=False, value=...)
    assert date_picker.value == [..., ...]
```

In general it's fine to be less DRY and more explicit when writing tests than writing source code. e.g. `[v.date() for v in value_to_date]` is nice and general and would work for many cases but is less clear than just spelling out the two values you want here explicitly.